ir/dict.go

```diff
@@ -139,16 +140,15 @@ func (d Dict) CopySetTag(tag string, key Node, value Node) Dict {
 	return c
 }
 
-func (d *Dict) Remove(key Node) Node {
+func (d *Dict) Remove(key Node) bool {
```

Comment:
Changing returning value to prevent `Pair` from having to be a `Node`. We could also return directly a `Pair` instead of a `Node` if we want to return the removed `Pair`.

Reply:
This is my mistake in the original code. What should be returned is the deleted Value (which is a Node), or nil if nothing was deleted.